app/controllers/two_factor_authentication/personal_key_verification_controller.rb

```diff
@@ -24,6 +25,12 @@ def create
 
     private
 
+    def personal_key_allowed?
```

Comment:
can we name this as an action? From the name I'd guess it would be a simple predicate. Maybe `check_personal_key_enabled`